### DISCUSSION TITLE
Question on S3M specification

### DISCUSSION THREAD

**Discussion by firolightfog** [schismtracker/schismtracker#443] [Q&A]
Hi,

Sorry for the offtopic. I know it's not directly related to SchismTracker but I secrectly hope to find an answer here. 

According to the [S3M specification](https://github.com/lclevy/unmo3/blob/master/spec/s3m.txt) on the Order List the relevant part starts at position 97:
![image](https://user-images.githubusercontent.com/34127628/70309569-1c33ff80-180e-11ea-8ed2-ef53a370f2ee.png)

Then seeing this in a song: 
![image](https://user-images.githubusercontent.com/34127628/70309710-661ce580-180e-11ea-85e4-f6d8ec13d7ac.png)

I expected something like this:

> {0,1,2,1,255,255,255,255,255,255,255,255,255,255,255,255}

But in my test file I found this:

> {0,1,2,1,255,255,9,0,14,0,39,39,39,39,39,39}

Why? Any suggestion, guidance, hint would be much appreciated! 

Regards, Andras

**Comment by firolightfog** [comment]
Oh, I think I know. It's the "Parapointers to instruments" starting after a single 0xff (or a double 0xff), isn't it?
Cheers, A.

**Comment by jangler** [comment]
The number of bytes in "Orders" equals the OrdNum value in the header file. 0xff just means an empty order (`---`); the order values can contain arbitrarily many consecutive 0xff's without ending.

**Comment by firolightfog** [comment]
This is the hex view of a simple S3M (with no patterns):

![S3M file clarification](https://user-images.githubusercontent.com/34127628/70379754-c4da8000-1930-11ea-84be-10277505b717.png)

**Comment by jangler** [comment]
Note the values in the red outlined section, and note the addresses of the AdLib instruments. See the connection?

**Comment by jangler** [comment]
And after the instrument pointers, the rest are pattern data pointers.

**Comment by firolightfog** [comment]
Oh, I see now! It's all there: 09, 0e,13! 
Many thanks, @jangler!
